File: Java/src/1720_DecodeXORedArray/DecodeXORedArray.java
class Solution {
    public int[] decode(int[] encoded, int first) {
        int undecoded[]=new int[encoded.length+1];
            
            undecoded[0]=first;
            
            int j=0;
        
            for(int i=1;i<undecoded.length;i++)
            {
                undecoded[i]=undecoded[i-1]^encoded[j];
                j++;
            }
        return undecoded;
    }
}

File: golang/1720_DecodeXORedArray/decodeXORedArray.go
package decodexoredarray

func decode(encoded []int, first int) []int {
	undecoded := []int{}
	undecoded = append(undecoded, first)
	for i := 0; i < len(encoded); i++ {
		undecoded = append(undecoded, undecoded[i]^encoded[i])
	}
	return undecoded
}
